packages/solid/store/src/modifiers.ts

```diff
@@ -110,7 +110,7 @@ export function reconcile<T extends U, U>(
   const { merge, key = "id" } = options,
     v = unwrap(value);
   return state => {
-    if (!isWrappable(state) || !isWrappable(v)) return v;
+    if (!isWrappable(state) || !isWrappable(v)) return v as T;
```

Comment:
TypeScript type guards fail here, probably in the next 10 years they'll fix this.